prov/peer: Introducing peer "provider" with peer_cq to begin with

grom72 opened the pull request:
include/ofi_peer.h file along with prov/peer directory defines space where peer-related common code is placed.
peer_cq is the first structure that has been defined. It is based on the coll_cq structure. coll provider uses peer_cq instead of removed coll_cq.

See [man/fi_peer.3.md](man/fi_peer.3.md)

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pmem/libfabric/4)
<!-- Reviewable:end -->

grom72 committed:
prov/peer: Introducing peer "provider" with peer_cq to begin with

include/ofi_peer.h file along with prov/peer directory defines space
where peer-related common code is placed.
peer_cq is the first structure that has been defined. It is based on
coll_cq structure. coll provider uses peer_cq instead of removed coll_cq.

See man/fi_peer.3.md

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

ldorau reviewed (changes requested):
> Reviewed 7 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmem/libfabric/4)* status: all files reviewed, 13 unresolved discussions (waiting on @grom72)

___
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIILm-_0RBeqXTG7BRr):*
introducing (small letter)



_Code quote:_
```
Introducing
```
___
*[`-- commits` line 5 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIILzUT5XttAny3l44Q):*
peer-related



_Code quote:_
```
peer related
```
___
*[`-- commits` line 9 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIIMAS64a2PAA6Bmu3w):*
`See man/fi_peer.3.md.` - the commit message is in the plain-text format, not MD.



_Code quote:_
```
See [man/fi_peer.3.md](man/fi_peer.3.md)
```
___
*[`Makefile.am` line 89 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIIMWua9s_Lo6uSuQ6y:-NIIMWua9s_Lo6uSuQ6z:b-sgyo72) ([raw file](https://github.com/pmem/libfabric/blob/c459fc39d195c35eba59c6c419215cd5066f8186/Makefile.am#L89)):*
> ```Makefile
> 	prov/util/src/cuda_ipc_monitor.c \
> 	prov/peer/src/peer_cq.c 	\
> 	prov/peer/src/peer.h 	        \
> ```

too many spaces or tabs (use only tabs, not spaces)



_Code quote:_
```Makefile
peer.h 	        \
```
___
*[`include/ofi_peer.h` line 2 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIIMt6D9Ml8nsNUqpGN:-NIIMt6E6hLUz8uv8izi:b-hg7te1) ([raw file](https://github.com/pmem/libfabric/blob/c459fc39d195c35eba59c6c419215cd5066f8186/include/ofi_peer.h#L2)):*
> ```objc
> /*
>  * Copyright (c) 2019-2022 Intel Corporation, Inc.  All rights reserved.
> ```

Why not just 2022?



_Code quote:_
```objc
2019-2022
```
___
*[`include/ofi_peer.h` line 40 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIIN4i8ApvUfK1c4h3T:-NIIN4i8ApvUfK1c4h3U:b3ybwik) ([raw file](https://github.com/pmem/libfabric/blob/c459fc39d195c35eba59c6c419215cd5066f8186/include/ofi_peer.h#L40)):*
> ```objc
> #include <ofi_list.h>
> #include <ofi_atom.h>
> #include <ofi_bitmask.h>
> ```

Are they really required here?



_Code quote:_
```objc
#include <ofi_atom.h>
#include <ofi_bitmask.h>
```
___
*[`include/ofi_peer.h` line 45 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIIN9RsEVgumvZMnVFr:-NIIN9RsEVgumvZMnVFs:b-ua1qv7) ([raw file](https://github.com/pmem/libfabric/blob/c459fc39d195c35eba59c6c419215cd5066f8186/include/ofi_peer.h#L45)):*
> ```objc
> 
> 
> int ofi_peer_cq_open(struct fid_domain *domain, struct fi_cq_attr *attr,
> ```

Too many empty lines in between.



_Code quote:_
```objc
#include <ofi_util.h>



int ofi_peer_cq_open(struct fid_domain *domain, struct fi_cq_attr *attr,
```
___
*[`include/ofi_peer.h` line 52 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIIOpKwB-rAUkkTR63i:-NIIOpKwB-rAUkkTR63j:bxupm5j) ([raw file](https://github.com/pmem/libfabric/blob/c459fc39d195c35eba59c6c419215cd5066f8186/include/ofi_peer.h#L52)):*
> ```objc
> 			fi_addr_t src);
> 
> #endif // _OFI_PEER_H_
> ```

`/* _OFI_PEER_H_ */` - `//` is a C++ comment



_Code quote:_
```objc
// _OFI_PEER_H_
```
___
*[`prov/coll/src/coll_coll.c` line 726 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIIQ7jX8xB3IVEJ8Lyf:-NIIQ7jX8xB3IVEJ8Lyg:b-pygwnj) ([raw file](https://github.com/pmem/libfabric/blob/c459fc39d195c35eba59c6c419215cd5066f8186/prov/coll/src/coll_coll.c#L726)):*
> ```C
> 	ep = container_of(coll_op->ep, struct coll_ep, util_ep.ep_fid);
> 
> 	ret = ofi_peer_cq_write(&ep->util_ep.tx_cq->cq_fid, coll_op->context, 
> ```

redundant space at the end



_Code quote:_
```C
,·
```
___
*[`prov/peer/src/peer.h` line 39 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIIQl696BV9yp2lvScw:-NIIQl696BV9yp2lvScx:bsioia4) ([raw file](https://github.com/pmem/libfabric/blob/c459fc39d195c35eba59c6c419215cd5066f8186/prov/peer/src/peer.h#L39)):*
> ```objc
> #if HAVE_CONFIG_H
> #include <config.h>
> #endif
> ```

What is it for?



_Code quote:_
```objc
#if HAVE_CONFIG_H
#include <config.h>
#endif
```
___
*[`prov/peer/src/peer.h` line 43 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIIQbTdDrH-DZ0wuT8Q:-NIIQbTdDrH-DZ0wuT8R:b-nsqcu9) ([raw file](https://github.com/pmem/libfabric/blob/c459fc39d195c35eba59c6c419215cd5066f8186/prov/peer/src/peer.h#L43)):*
> ```objc
> #include <ofi.h>
> #include <ofi_util.h>
> ```

redundant empty line
___
*[`prov/peer/src/peer_cq.c` line 99 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIIRMpN7r9VNfVc8404:-NIIRMpN7r9VNfVc8405:b44s5rm) ([raw file](https://github.com/pmem/libfabric/blob/c459fc39d195c35eba59c6c419215cd5066f8186/prov/peer/src/peer_cq.c#L99)):*
> ```C
> 
> 	ret = ofi_cq_init(provider, domain, attr, &cq->util_cq, 
> 		&ofi_cq_progress, NULL);
> ```

Doesn't it fit into one line?



_Code quote:_
```C
	ret = ofi_cq_init(provider, domain, attr, &cq->util_cq,
		&ofi_cq_progress, NULL);
```
___
*[`prov/peer/src/peer_cq.c` line 122 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIIRcOZC_Wuij3ZqQ9t:-NIIRcOZC_Wuij3ZqQ9u:b44s5rm) ([raw file](https://github.com/pmem/libfabric/blob/c459fc39d195c35eba59c6c419215cd5066f8186/prov/peer/src/peer_cq.c#L122)):*
> ```C
> ssize_t ofi_peer_cq_write(struct fid_cq *cq_fid, void *context, uint64_t flags,
> 			size_t len, void *buf, uint64_t data, uint64_t tag,
> 			fi_addr_t src)
> ```

Doesn't it fit into one line?



_Code quote:_
```C
			size_t len, void *buf, uint64_t data, uint64_t tag,
			fi_addr_t src)
```


<!-- Sent from Reviewable.io -->

grom72 reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/pmem/libfabric/4)* status: 3 of 7 files reviewed, 13 unresolved discussions (waiting on @ldorau)

___
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIILm-_0RBeqXTG7BRr):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

introducing (small letter)
</blockquote></details>

No - see Sean's info: https://github.com/ofiwg/libfabric/pull/8278#discussion_r1036625137
___
*[`-- commits` line 5 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIILzUT5XttAny3l44Q):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

peer-related
</blockquote></details>

Done.
___
*[`-- commits` line 9 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIIMAS64a2PAA6Bmu3w):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

`See man/fi_peer.3.md.` - the commit message is in the plain-text format, not MD.
</blockquote></details>

Done.
___
*[`Makefile.am` line 89 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIIMWua9s_Lo6uSuQ6y:-NIJMfgWCyt4luaRWiiA:b-896fix) ([raw file](https://github.com/pmem/libfabric/blob/c459fc39d195c35eba59c6c419215cd5066f8186/Makefile.am#L89)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

too many spaces or tabs (use only tabs, not spaces)
</blockquote></details>

Done.
___
*[`include/ofi_peer.h` line 2 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIIMt6D9Ml8nsNUqpGN:-NIJMkXO05GIGZgpulGi:b-896fix) ([raw file](https://github.com/pmem/libfabric/blob/c459fc39d195c35eba59c6c419215cd5066f8186/include/ofi_peer.h#L2)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

Why not just 2022?
</blockquote></details>

Done.
___
*[`include/ofi_peer.h` line 40 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIIN4i8ApvUfK1c4h3T:-NIJMlef0cOoBir2uErn:b-896fix) ([raw file](https://github.com/pmem/libfabric/blob/c459fc39d195c35eba59c6c419215cd5066f8186/include/ofi_peer.h#L40)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

Are they really required here?
</blockquote></details>

Done.
___
*[`include/ofi_peer.h` line 45 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIIN9RsEVgumvZMnVFr:-NIJMmM399stGzRiT9oM:b-896fix) ([raw file](https://github.com/pmem/libfabric/blob/c459fc39d195c35eba59c6c419215cd5066f8186/include/ofi_peer.h#L45)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

Too many empty lines in between.
</blockquote></details>

Done.
___
*[`include/ofi_peer.h` line 52 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIIOpKwB-rAUkkTR63i:-NIJMmn41gXAvMaonqmT:b-896fix) ([raw file](https://github.com/pmem/libfabric/blob/c459fc39d195c35eba59c6c419215cd5066f8186/include/ofi_peer.h#L52)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

`/* _OFI_PEER_H_ */` - `//` is a C++ comment
</blockquote></details>

Done.
___
*[`prov/coll/src/coll_coll.c` line 726 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIIQ7jX8xB3IVEJ8Lyf:-NIJMpYh28Osr3SKQgg6:b-896fix) ([raw file](https://github.com/pmem/libfabric/blob/c459fc39d195c35eba59c6c419215cd5066f8186/prov/coll/src/coll_coll.c#L726)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

redundant space at the end
</blockquote></details>

Done.
___
*[`prov/peer/src/peer.h` line 39 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIIQl696BV9yp2lvScw:-NIJLiWqB2m93OHpE-il:b-rwzsrv) ([raw file](https://github.com/pmem/libfabric/blob/c459fc39d195c35eba59c6c419215cd5066f8186/prov/peer/src/peer.h#L39)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

What is it for?
</blockquote></details>

I do not know yet.



_Code quote (from prov/coll/src/coll_coll.c):_
```C
ofi_peer_cq_write
```
___
*[`prov/peer/src/peer.h` line 43 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIIQbTdDrH-DZ0wuT8Q:-NIJMt6g74WaFGzcrPNZ:b-896fix) ([raw file](https://github.com/pmem/libfabric/blob/c459fc39d195c35eba59c6c419215cd5066f8186/prov/peer/src/peer.h#L43)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

redundant empty line
</blockquote></details>

Done.
___
*[`prov/peer/src/peer_cq.c` line 99 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIIRMpN7r9VNfVc8404:-NIJLzs78pXWQweE9Kj0:b32w12) ([raw file](https://github.com/pmem/libfabric/blob/c459fc39d195c35eba59c6c419215cd5066f8186/prov/peer/src/peer_cq.c#L99)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

Doesn't it fit into one line?
</blockquote></details>

No
___
*[`prov/peer/src/peer_cq.c` line 122 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIIRcOZC_Wuij3ZqQ9t:-NIJM4eu4_yBSWMLz8Eu:b-i5dm5r) ([raw file](https://github.com/pmem/libfabric/blob/c459fc39d195c35eba59c6c419215cd5066f8186/prov/peer/src/peer_cq.c#L122)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

Doesn't it fit into one line?
</blockquote></details>

No - we have only 80 characters


<!-- Sent from Reviewable.io -->

grom72 committed:
prov/peer: add provider parameter to peer_cq_init

Adding struct fi_provider* as the first parameter to peer_cq_init()
makes it similar to ofi_cq_init().

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

vidsouza committed:
prov/efa: Do no ignore hints->domain_attr->name

Prior to this patch, the provider would ignore the domain
name in hints->domain_attr->name even if this was set.
This patch fixed this issue.

Signed-off-by: Vishwas Dsouza <vidsouza@amazon.com>

vidsouza committed:
fabtests/pytest/efa: Add efa device selection test to fabtests suite

This test tests the -d <domain_name> argument of the fabtests.
All the EFA domain names are tested with this test with the
corresponding fabtest.

Signed-off-by: Vishwas Dsouza <vidsouza@amazon.com>

ldorau reviewed (changes requested):
> Reviewed 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmem/libfabric/4)* status: all files reviewed, 1 unresolved discussion (waiting on @grom72)

___
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/pmem/libfabric/4#-NIILm-_0RBeqXTG7BRr):*
<details><summary><i>Previously, grom72 (Tomasz Gromadzki) wrote…</i></summary><blockquote>

No - see Sean's info: https://github.com/ofiwg/libfabric/pull/8278#discussion_r1036625137
</blockquote></details>

OK
___
*[`prov/peer/src/peer_cq.c` line 92 at r2](https://reviewable.io/reviews/pmem/libfabric/4#-NIWO98lBS40b2beBBHg:-NIWO98lBS40b2beBBHh:bsnnk5t) ([raw file](https://github.com/pmem/libfabric/blob/7c323c43f930cb8377ff1aefd9992e193cdc2d6f/prov/peer/src/peer_cq.c#L92)):*
> ```C
> 	cq->peer_cq = peer_context->cq;
> 
> 	ret = ofi_cq_init(provider, domain, attr, &cq->util_cq, 
> ```

Redundant space at the end



_Code quote:_
```C
,·
```


<!-- Sent from Reviewable.io -->

haichangsi reviewed (commented):
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmem/libfabric/4)* status: all files reviewed, 1 unresolved discussion (waiting on @grom72)



<!-- Sent from Reviewable.io -->

grom72 committed:
prov/peer: Minimalist version of peer_cq

struct peer_cq is not used at all. All operation are based on fid_peer_cq.
There are no real CQ on util_coll. Only fid_peer_cq is binded to util_coll_ep.

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

grom72 committed:
prov/rxm: remove util_coll_cq from rxm

util_coll_cq is no longer needed as we use only fid_peer_cq to link util_coll_ep and
rxm_ep.util_cq directly with help of bind() operation.

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

grom72 committed:
prov/peer: Remove peer_cq implementation

struct peer_cq is no longer needed.

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>

ldorau reviewed (commented):
> Reviewed 1 of 9 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pmem/libfabric/4)* status: 3 of 11 files reviewed, 1 unresolved discussion (waiting on @grom72 and @haichangsi)



<!-- Sent from Reviewable.io -->

wzamazon committed:
prov/efa: add a FI_WARN about endpoint address after it is created

This patch added a call to FI_WARN to print endpoint's address
and libfabric version after endpoint is created.

Signed-off-by: Wei Zhang <wzam@amazon.com>
